### PR TITLE
Milestone 5 friendships

### PR DESCRIPTION
**Pull request by DanteAlonsoHT**
<h1>Social media website - Milestone: friendships v1</h1>

In this pull request, I have been working on the friendships among users,  any user is capable to request a friendship from another user, and this another user can accept or reject the request if he/she wants to do that, also we have included some many features to put more privacy inside the app according to what project requirements expect about this milestone, we've been working on the next files:

- Models for users
- Controller for posts and users
- Helpers for users and friendships
- Views for users

Finally, I got the approvement according to rubocop linters :)

### CODE REVIEW

**Review by bini-i — CHANGES_REQUESTED** [review]
# Status: Changes Required :yellow_circle:            
### Hello @DanteAlonsoHT and @Fondem-Jr  :wave: 

Your project looks good and meets almost all the requirements:
#### To highlight
- descriptive pull request
- good use of gitflow

#### Please work on the following missing requirements and submit for review. 
#### To improve
- Kindly make sure that a user sees posts created by the user and friends(users with approved request) only as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/friends_timeline.png).
- Please make sure friend requests made to a user are not visible from another user, currently I can see who made requests to user1 while logged in as user3 as shown below:
image
- Kindly include unit and integration tests as specified [here](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png).
- Please customize your README and edit sections like the Author section, to make your repo more professional and allow visitors of your repo to engage with you.

#### If you have any question feel free to ask within the PR and ping me on slack @Binyam Hailemeskel

### HAPPY CODING :computer: :blush: 

> #### Reviewed by Binyam Hailemeskel (Code Reviewer)
> GitHub: [@bini-i](https://github.com/bini-i) 
> Twitter: [@binyamshewa](https://twitter.com/binyamshewa)
> LinkedIn: [LinkedIn](https://www.linkedin.com/in/binyam-hailemeskel-728048151/)#

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+      </p>
+    <% end %>
+  </div>
+  <% end %>
+
+  <% if current_user.name == @user.name %>
+  <h4 class="my-5"> Welcome! you're in your own profile </h4>
+  <% else %>
+  <h4 class="my-5"><%= "You're in #{@user.name} profile" %> <span> <%= invite_or_pending_btn(@user) %></span></h4>
+  <% end %>
```
Comment (rindrajosia) [inline app/views/users/show.html.erb]: [OPTIONAL]Make sure that there no logic in your view.

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- User can invite friends :+1: 
- User can see friendships requests :100: 
- Posts timeline includes posts written by logged-in user and their friends :1st_place_medal: 

To improve :factory_worker: [REQUIRED]:

- [x] Please add more test units. Don't forget to add negative tests for each case. The aim is to find if your APP work as expected.

Nice to have [OPTIONAL] :

- [ ] It will be nice if user_id and post_id inside comments, likes, and posts tables are also a foreign key.
- [x] Make sure that there no logic in your view.

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, thank you for working on changes suggested by previous reviewers :ok_hand:
## To Highlight 🎉
- [x] Git flow is followed properly.
- [x] README includes all the instructions to run the project locally.

## To Improve 🙏
- [x] Linter checks are failing in PR. Please fix that.

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Review by BrianSammit — APPROVED** [review]
STATUS: APPROVED 🟢 👏 
--------------
Hello @DanteAlonsoHT, great job you did on this milestone 💪 good job working on the required changes 💯 
------------------
Now you can merge this branch 🥇 
HAPPY CODING!! AND GOOD LUCK WITH THE COMMING MILESTONES 🍀 🤞 

kvUU7

**Pull request merged**